scripts/cmake/vcpkg_configure_make.cmake

```
@@ -273,7 +286,15 @@ function(vcpkg_configure_make)
         else()
             z_vcpkg_append_to_configure_environment(configure_env CPP "${VCPKG_DETECTED_CMAKE_C_COMPILER} -E")
             z_vcpkg_append_to_configure_environment(configure_env CC "${VCPKG_DETECTED_CMAKE_C_COMPILER}")
-            z_vcpkg_append_to_configure_environment(configure_env CC_FOR_BUILD "${VCPKG_DETECTED_CMAKE_C_COMPILER}")
+            if(NOT VCPKG_CROSSCOMPILING)
+                z_vcpkg_append_to_configure_environment(configure_env CC_FOR_BUILD "${VCPKG_DETECTED_CMAKE_C_COMPILER}")
+                z_vcpkg_append_to_configure_environment(configure_env CPP_FOR_BUILD "${VCPKG_DETECTED_CMAKE_C_COMPILER} -E")
+                z_vcpkg_append_to_configure_environment(configure_env CXX_FOR_BUILD "${VCPKG_DETECTED_CMAKE_CXX_COMPILER}")
+            else()
+                z_vcpkg_append_to_configure_environment(configure_env CC_FOR_BUILD "touch a.out | touch conftest${VCPKG_HOST_EXECUTABLE_SUFFIX} | true")

```

Is this trying to make configure checks happy and then hope for the best in the build or is there something more to it?

---

This is mainly to make configure happy and make the build work how it is intended to work within vcpkg. (i.e. no build of host stuff)

If you want your port to actually build you need the following:
+ declare host dependency on itself
+ copy/install the (internal) binaries/tools to a location in `CURRENT_PACKAGES_DIR/(manual-)tools/PORT(/bin)` 
+ put the tools on PATH if VCPKG_CROSSCOMPILING. (Make rules typically simply invoke by name which is why this works.)

Since these steps are individual to a port it cannot be generally handled in `vcpkg_configure_make`.

---

I wonder if the default behaviour shouldn't be failure, and the fake success should be activated by an option.
I also wonder if it is really portable to use `|` as part of `CC_FOR_BUILD` because it assumes interpretation by a shell.

---

> I wonder if the default behaviour shouldn't be failure, and the fake success should be activated by an option.
I also wonder if it is really portable to use | as part of CC_FOR_BUILD because it assumes interpretation by a shell.

If it ever becomes a problem it can be replaced with a true shell script doing the same ;)

---

My first point is that such variables have different interpretations. Usually it is okay to supply a command with arguments. But you proposal is to supply a command line with pipe operator. It must be interpreted by a shell, it cannot be passed directly to the system. 

My second point is that I actually want to be made aware of a build process which tries to access for CC_FOR_HOST. I don't want the build to silently succeed, only to discover strange effects at runtime.

So it may work for `gmp` right now. But this shouldn't be generalized.

---

> It must be interpreted by a shell, it cannot be passed directly to the system.

As I said, the cmd line could be replaced with a simple script doing the same if it ever is a problem. (configure already as to run in a shell, so? And all Makefiles generated by automake or manually written will contain some sort of piping)

> My second point is that I actually want to be made aware of a build process which tries to access for CC_FOR_BUILD.
 I don't want the build to silently succeed, only to discover strange effects at runtime.

it won't build anything. It just making configure happy. vcpkg should never try to build anything for the host triplet unless it is a target=host build. This is why this is necessary. Otherwise you run into the case where it silently accepts whatever CC is which might or might not work which is surely incorrect. 

Currently crosscompiling with `vcpkg_configure_make` is basically unsupported. With this PR it will correctly deactivate `_FOR_BUILD` which will run into a build failures if the required binaries build by `_FOR_BUILD` and required by the build are not invokeable. Since these internal binaries do typically not have install rules there is little risk that those are available from the system without any extra work by vcpkg.